File: string/550-similarPairs.py
class Solution:
    def similarPairs(self, words: List[str]) -> int:
        # n=len(words)
        # ans=0
        # for i in range(n):
        #     for j in range(i+1,n):
        #         if Counter(words[i]).keys()==Counter(words[j]).keys():
        #             ans+=1
        # return ans
        # n=len(words)
        # ans=0
        # for i in range(n):
        #     s=set((words[i]))
        #     for j in range(i+1,n):
        #         ss=set((words[j]))
        #         if s==ss:
        #             ans+=1
        # return ans
        def id(x):
            return "".join(sorted(set(x)))
        ans=0
        c=Counter()
        for w in words:
            x=id(w)
            ans+=c[x]
            c[x]+=1
        return ans

File: string/550-similarPairs.cpp
#include<string>
#include<iostream>
#include<map>
#include<set>
#include<vector>
using STR=std::string;
using MAP=std::map<STR,int>;
using SET=std::set<char>;
using VEC=std::vector<STR>;
using std::cout;
STR str(const STR& s)
{
    SET seen{};
    STR tmp{""};
    for(auto &c:s)
        seen.insert(c);
    for(auto &cc:seen)
        tmp+=cc;
    return tmp;
}

int similarPairs(const VEC& words)
{
    MAP C{};
    int ans=0;
    for(auto &w:words)
    {
        STR t{str(w)};
        ans+=C[t];
        C[t]+=1;
    }
    return ans;
}

int main()
{
    VEC words{"aabb","ab","abc","bcaaa"};
    cout<<similarPairs(words);
}